### PR TITLE
[Communication Services - Sip routing] PATCH /sip routing new version API swagger for review

### LINKED ISSUE #19503: [Communication Services - Sip routing] REST API Review
Service name: Communication
Key contact for the review: Vadym Moskalenko
Whether this is a new or existing API: existing API (new version)
Whether or not your service already been deployed: Service has been deployed
A brief description of the material to be reviewed/changed: New swagger file definition to review for a new API version of PATCH /sip routing

Pull Request:
https://github.com/Azure/azure-rest-api-specs/pull/19502

Link to Design Document: https://microsoft.sharepoint-df.com/:w:/t/IC3SDK/ESvU8UOBK-ZMn_2frVj_WLcB50ssGwrGZmCp4v-Fn-C-aQ?e=m0O2hg

### PR DESCRIPTION
**Pull request by nostojicMs**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

fix https://github.com/Azure/azure-rest-api-specs/issues/19503

### CODE REVIEW

**Comment by nostojicMs** [comment]
Hello @bexxx and @tg-msft kindly reminding you to take a look at this PR. 
Related issue to this PR: https://github.com/Azure/azure-rest-api-specs/issues/19503
Thank you

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    "version": "2022-09-01-preview"
+  },
+  "paths": {
+    "/sip": {
+      "get": {
+        "tags": [
+          "SipConfiguration"
+        ],
+        "summary": "Gets SIP configuration for resource.",
+        "operationId": "GetSipConfiguration",
```
Comment (bexxx) [inline specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:18]: >GetSipConfiguration [](http://example.com/codeflow?start=24&length=19)

it's recommended to use {collection}_{meaningful verb}
like SipConfiguration_Get.

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            },
+            "x-ms-error-response": true
+          },
+          "500": {
+            "description": "All other api errors.",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
```
Comment (bexxx) [inline specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:100]: should be covered by the "default'

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+      "type": "string",
+      "description": "Type of domain",
+      "enum": [
+        "Custom",
+        "Azure"
+      ],
+      "x-ms-enum": {
+        "name": "DomainType",
+        "modelAsString": false
+     }
```
Comment (bexxx) [inline specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:265]: add description for the different values.

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+            "$ref": "#/definitions/TrunkRoute"
+          }
+        }
+      }
+    },
+    "DomainType": {
+      "type": "string",
+      "description": "Type of domain",
+      "enum": [
+        "Custom",
```
Comment (bexxx) [inline specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:259]: >C [](http://example.com/codeflow?start=9&length=1)

the casing might be result of your implementation.
Do you want to document azure and custom or really Azure and Custom?

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json (hunk 1 above)
Comment (nostojicMs) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json]: Renamed it. Thx

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json (hunk 2 above)
Comment (nostojicMs) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:100]: Default is stated bellow this, on line 101. Please check if that is what you are referring to. Thx

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+          "description": "Trunk routes for routing calls.",
+          "maxLength": 250,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TrunkRoute"
+          }
+        }
+      }
+    },
+    "SipConfigurationPatch": {
```
[the remark this answers is not in the document]
Comment (nostojicMs) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:226]: It represents object of PATCH method, so I would leave it as ...Patch. Thx

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json (hunk 4 above)
Comment (nostojicMs) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json]: Changed casing to lower. Thx

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+          "maxLength": 250,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "Domain": {
+      "description": "Represents Domain.",
```
[the remark this answers is not in the document]
Comment (nostojicMs) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json]: Done. Thx

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json (hunk 1 above)
Comment (nostojicMs) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json]: @bexxx seems like this renaming of operation id is causing breaking change. I would recommend putting it back to old value so we avoid introducing breaking change.  Do you agree?

Comment (bexxx) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json]: This depends. If you have an SDK published this might affect more customers, if you just have preview versions out and not SDK, it might be fine to fix it now before this service gets more mature. All that is needed for a breaking change is the sign off in the breaking change review. so this depends on your decision, both options are fine.

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json (hunk 2 above)
Comment (bexxx) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:100]: That is my point. You might want to remove all obvious error responses above. There is e.g. little value in documenting 500 response in addition to default.

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json (hunk 3 above)
Comment (bexxx) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:264]: i meant as ms extension like this:

"x-ms-enum": {
              "name": "Purpose",
              "modelAsString": true,
              "values": [
                {
                  "value": "fine-tune",
                  "description": "This file contains training data for a fine tune job."
                },
                {
                  "value": "fine-tune-results",
                  "description": "This file contains the results of a fine tune job."
                }
              ]
            }

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json (hunk 1 above)
Comment (nostojicMs) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json]: Renamed. Thank you

**Comment by nostojicMs** [comment]
This new version of Sip Routing API introduces breaking change by adding required parameter "Enabled" in response as part of "Trunks" and "Domains" in GET and PATCH methods. This change is introduced to show if Trunk and corresponding Domain is used (Enabled=true) in Sip configuration. Domain validation will be performed only to enabled Domains. If Domain is not enabled (Enabled=False) it will not be used in sip configuration. Trunks can be enabled only if corresponding Domain is enabled. 
Change of operation Ids in GET and PATCH are changed to follow recommended guidelines for constructing swagger. Since this is not still GA it's good opportunity to do this change.

cc PR assignee @jhendrixMSFT

**Comment by nostojicMs** [comment]
Hi, here's the link to the breaking change intake of renaming of operation id: https://msazure.visualstudio.com/One/_workitems/edit/15324575

File: specification/communication/data-plane/SipRouting/readme.ruby.md [hunk 7]
```diff
@@ -7,6 +7,10 @@ package-name: azure_mgmt_communicationservices
 package-version: 2021-05-01-preview
 azure-arm: true
 ```
+### Tag: package-2022-09-01-preview and ruby
+
+These settings apply only when `--tag=package-2022-09-01-preview --ruby` is specified on the command line.
+Please also specify `--ruby-sdks-folder=<path to the root directory of your azure-sdk-for-ruby clone>`.
```
Comment (tg-msft) [inline specification/communication/data-plane/SipRouting/readme.ruby.md:13]: Do you need settings here?

**Review by tg-msft — APPROVED** [review]
The changes look good to the API Stewardship Board.

File: specification/communication/data-plane/SipRouting/readme.ruby.md (hunk 7 above)
Comment (nostojicMs) [reply specification/communication/data-plane/SipRouting/readme.ruby.md:8]: Hello, I put those settings for new version so it's possible to generate each version. Does that make sense?
